### PR TITLE
Add vuid 04813 04814, Validate video buffer usage

### PR DESCRIPTION
**Pull request by ziga-lunarg**

### CODE REVIEW

File: layers/buffer_validation.cpp [hunk 1]
```diff
@@ -4802,6 +4802,36 @@ bool CoreChecks::PreCallValidateCreateBuffer(VkDevice device, const VkBufferCrea
...
+                }
+            }
+        }
+        if (!has_decode_codec_operation) {
+            skip |= LogError(device, "VUID-VkBufferCreateInfo-usage-04813",
+                             "vkCreateBuffer(): pCreateInfo->usage is %s, but pNext chain does not include VkVideoProfilesKHR with "
+                             "a decode codec-operation.",
+                             string_VkBufferUsageFlags(pCreateInfo->usage).c_str());
+        }
+        if (!has_encode_codec_operation) {
```
Comment (Tony-LunarG) [inline layers/buffer_validation.cpp:4827]: Doesn't pCreateInfo->usage have to have VK_BUFFER_USAGE_VIDEO_ENCODE_SRC_BIT_KHR, VK_BUFFER_USAGE_VIDEO_ENCODE_DST_BIT_KHR for 04814 to apply?  The above only checks for DECODE in usage.

Comment (ziga-lunarg) [reply layers/buffer_validation.cpp:4827]: Not sure I am understanding. The check for `VK_BUFFER_USAGE_VIDEO_ENCODE_SRC_BIT_KHR` or `VK_BUFFER_USAGE_VIDEO_DECODE_DST_BIT_KHR` is in line 4805. This if here checks if `videoCodecOperation` includes a encode codec-operation.

Comment (Tony-LunarG) [reply layers/buffer_validation.cpp:4827]: The only checks I see in 4805 are for DECODE_SRC or DECODE_DST, no ENCODE anything.  Or am I missing something?

Comment (ziga-lunarg) [reply layers/buffer_validation.cpp:4836]: You're right, I got confused.

**Review by Tony-LunarG — APPROVED** [review]
LGTM

**Pull request merged**